### DISCUSSION TITLE
Filtering triangle on multiple indexes

### DISCUSSION THREAD

**Discussion by Obalus2022** [casact/chainladder-python#318] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/110035421/181641313-c6d9520f-72a5-4e42-a10f-ef36b97e8882.png)

This is from the Triangles tutorial.  How do you filter when want to search mulltiple indexes, say GrpName =AEGIS AND lob=GL?

Thanks in advance

**Comment by jbogaardt** [comment, marked as answer]
Interesting we don't talk about that but could be useful for people new to chainladder and new to pandas.  

It would be the same as you would in `pandas`. You can chain any number of boolean expressions together to subset your data.

```python
clrd[(clrd['GRNAME']=='Aegis Grp')&(clrd['LOB']=='othliab')]
```

**Comment by Obalus2022** [comment]
Thanks so much that was very helpful.  It throws me, because I think of this triangle as a 5-dimensional array, but I guess its still only a four dimensional one.  Is there recommended tutorials for pandas as it pertains to chainladder.  Same question for scikit.

**Comment by jbogaardt** [comment]
The CAS is hosting a [virtual workshop](https://www.casact.org/event/2022-cas-virtual-workshop-introduction-python-0) in a couple weeks that covers pandas, scikit-learn, and chainladder among other things in more detail. This workshop aims to reinforce python skills across all three packages, but it is not free.

For free, self-guided options, it's been a while since I've looked and there has been a ton of development. I can only assume that the [pandas getting started](https://pandas.pydata.org/docs/getting_started/index.html#getting-started) section is the most definitive resource.  Similarly, the [scikit-learn tutorials](https://scikit-learn.org/stable/tutorial/index.html#tutorial-menu) are also a reasonable place to start.  If someone newer to learning python has any thoughts, please comment.

**Comment by SRwcf** [comment]
Indeed, resources mentioned by @jbogaardt are great. Moreover, I have also found courses in online learning portals like coursera and udemy to be helpful for self-learning. Usually these courses are free or financial aid is available. 

Though there is scarcity of specific tutorials for pandas pertaining to ChainLadder, I found  https://www.udemy.com/share/1013ko/ [Data Analysis with Pandas and Python] course in udemy super helpful. Hope this helps.
